### ISSUE TITLE
Consider adding validation for release.sh 

### ISSUE THREAD

**Issue by mattmoor** [knative/test-infra#1614]
This came up in changes to generate-yamls.sh in Serving (https://github.com/knative/serving/pull/6431#issuecomment-571714135) , but this script  is unsuitable to ask developers to run locally because:
1. Some developers (like me) have broad enough permissions to actually cut a release and it isn't worth the risk.
2. The script also assumes that you have GCP access, which isn't a reasonable expectation.

tl;dr If we want presubmit validation of this file, then we should add it to a Prow job.

**Comment by adrcunha** [comment]
Note that while the basics of `generate-yamls.sh` is tested by the E2E tests, PR 6431 affects the output of a release, and how Knative Serving is installed by users, thus testing it (beyond E2E) is advisable to avoid broken releases.
